sycl/doc/design/CompileTimeProperties.md

```diff
@@ -806,26 +850,57 @@ into one (or both) of the following:
 In both cases, the decoration is a single **UserSemantic** decoration where the
 string literal is the same as the string literal in the LLVM annotation.
 
-When a SYCL structure member property needs to be represented in SPIR-V,
-however, we prefer to represent each property as an extended SPIR-V decoration
-rather than using a **UserSemantic** decoration.  There is no existing
-mechanism in the SPIR-V LLVM Translator to generate extended decorations like
-this, so we propose the following new mechanism.
+An exception to this is for a selection of FPGA-related decorations. If these
+are supported during translation from LLVM IR to SPIR-V the corresponding
+decorations will be generated, and otherwise it will fall back to creating a
+single **UserSemantic** decoration. In general these decorations occur in the
+annotation string as a series of **{X}** and **{X:Y}** where **X** is a reserved
+name and **Y** is one or more words and numbers separated by **,** or **:**,
+depending on the decoration.
+
+As such we propose an extension to this functionality with the following
+changes:
+
+* To bring it in line with the format of the metadata decorations, the parsing
+  of these decorations should allow the use of SPIR-V decoration identifiers
+  rather than reserved names. With this there need not be any agreement between
+  the translator and LLVM IR producer, as the identifiers are specified by the
+  SPIR-V specification.
+* For decorations parsed with decoration identifiers, only the **,** delimiter
+  is valid for separating decoration values.
+* In addition to words and numbers, string literals enclosed by quotation marks
+  are allowed as decoration values. No escapes are planned for this, so all
+  symbols between starting quotation mark and ending quotation mark are
+  considered part of the string literal.
 
 When a member variable property needs to be represented in SPIR-V, the
-`sycl-post-link` tool converts the `@llvm.ptr.annotation` intrinsic call into a
-call to the SPIR-V intrinsic `__spirv_AddMemberDecoration` which has a metadata
-function argument that specifies the decorations as illustrated below:
+`sycl-post-link` tool converts the `@llvm.ptr.annotation` intrinsic call
+produced by `[[__sycl_detail__::add_ir_annotations_member()]]` into another
+`@llvm.ptr.annotation` intrinsic call using this format. For example:
 
 ```
-%annotated_ptr = call i8* __spirv_AddMemberDecoration(i8* %ptr, metadata !0)
+; Contains decorations:
+;  * 7744 with no value.
+;  * 7745 with 20 and "str 1" as the values.
+@.str = private unnamed_addr constant [24 x i8] c"{7744}{7745:20,"str 1"}\00",
```

Comment:
I presume that `llvm-dis` would normally escape the internal double-quote characters in the disassembly.  For example something like:

```suggestion
@.str = private unnamed_addr constant [24 x i8] c"{7744}{7745:20,\"str 1\"}\00",
```

Reply:
Right you are! I have replaced `"` in the literal with `\22`, which is the way LLVM IR encodes it in string literals.